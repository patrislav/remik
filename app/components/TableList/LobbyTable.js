import React, {PropTypes} from 'react'
import styles from './LobbyTable.css'

const propTypes = {
  table: PropTypes.shape({
    id: PropTypes.string.isRequired
  }).isRequired
}

const LobbyTable = ({ table, ...props }) => (
  <div className={styles.wrapper} {...props}>
    Table #{table.id}
  </div>
)

LobbyTable.propTypes = propTypes

export default LobbyTable
