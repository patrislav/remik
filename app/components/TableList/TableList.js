import React, {PropTypes} from 'react'
import {connect} from 'react-redux'
import styles from './TableList.css'
import values from 'lodash/values'

import LobbyTable from './LobbyTable'

const propTypes = {
  tables: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string
  })),
  onJoin: PropTypes.func
}

const defaultProps = {
  tables: [],
  onJoin: () => {}
}

const mapStateToProps = state => ({
  tables: values(state.lobby.tables)
})

export const TableList = ({ tables, onJoin }) => (
  <section className={styles.section}>
    <ul className={styles.list}>
      {tables.map(table =>
        <li key={table.id} className={styles.item}>
          <LobbyTable table={table} onClick={() => onJoin(table.id)} />
        </li>
      )}
    </ul>
  </section>
)

TableList.propTypes = propTypes
TableList.defaultProps = defaultProps

export default connect(mapStateToProps)(TableList)
