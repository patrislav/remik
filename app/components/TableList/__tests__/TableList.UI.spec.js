import React from 'react'
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'
import times from 'lodash/times'

import {TableList} from '../TableList'

describe('<TableList />', () => {
  it('renders correct UI without props', () => {
    const wrapper = shallow(<TableList />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('renders correct UI with passed tables', () => {
    const tables = times(3, i => ({
      id: `table_${i}`
    }))
    const wrapper = shallow(<TableList tables={tables} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
