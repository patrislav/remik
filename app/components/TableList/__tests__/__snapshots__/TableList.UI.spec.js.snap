exports[`<TableList /> renders correct UI with passed tables 1`] = `
<section
  className="section">
  <ul
    className="list">
    <li
      className="item">
      <LobbyTable
        onClick={[Function]}
        table={
          Object {
            "id": "table_0",
          }
        } />
    </li>
    <li
      className="item">
      <LobbyTable
        onClick={[Function]}
        table={
          Object {
            "id": "table_1",
          }
        } />
    </li>
    <li
      className="item">
      <LobbyTable
        onClick={[Function]}
        table={
          Object {
            "id": "table_2",
          }
        } />
    </li>
  </ul>
</section>
`;

exports[`<TableList /> renders correct UI without props 1`] = `
<section
  className="section">
  <ul
    className="list" />
</section>
`;
